File: codejam/python/2010/1B/1B_FileFixIt.py
import sys
from collections import deque

## Input crap
infile = sys.stdin
intokens = deque()
def getTokens(): 
    while not intokens:
        for c in infile.readline().rstrip().split() :
            intokens.append(c)    
def gs(): getTokens(); return intokens.popleft()
def gi(): return int(gs())
def gf(): return float(gs())
def gbs(): return [c for c in gs()]
def gis(n): return [gi() for i in range(n)]
def ia(m): return [0] * m
def iai(m,v): return [v] * m
def twodi(n,m,v): return [iai(m,v) for i in range(n)]
def fill2(m) : r = gis(2*m); return r[0::2],r[1::2]
def fill3(m) : r = gis(3*m); return r[0::3],r[1::3],r[2::3]
def fill4(m) : r = gis(4*m); return r[0::4],r[1::4],r[2::4],r[3::4]
MOD = 998244353
##MOD = 1000000007

def main() :
    if len(sys.argv) > 1 : global infile; infile = open(sys.argv[1],'rt')
    ## PROGRAM STARTS HERE
    T = gi()
    d = {}
    for tt in range(1,T+1) :
        ## Do this sloppy and avoid the tree -- think the input is small enough for this to be fine
        N,M = gi(),gi(); l1 = [gs() for _ in range(N) ]; l2 = [gs() for _ in range(M) ]
        ll = set()
        for l in l1 :
            ll.add(l)
            for x in (l[:i] for i in range(1,len(l)) if l[i] == '/') : ll.add(x)
        n1 = len(ll)
        for l in l2 :
            ll.add(l)
            for x in (l[:i] for i in range(1,len(l)) if l[i] == '/') : ll.add(x)
        n2 = len(ll)
        ans = n2-n1
        print(f"Case #{tt}: {ans}")

if __name__ == "__main__" :
    main()

File: codejam/cpp/2010/1B/1B_FileFixIt.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<ll> vi;
typedef pair<ll,ll> pi;
#define FOR(i,a) for (ll i = 0; i < (a); i++)
#define len(x) (ll) x.size()
const ll INF = 1LL << 62;
const ll MOD = 1000000007;
//const ll MOD = 998244353;
const double PI = 4*atan(double(1.0));
vector<vector<ll>> twodi(ll N, ll M, ll v) { vector<vector<ll>> res(N); for (auto &vv : res) vv.resize(M,v); return res; }

int main (int argc, char **argv) {
    if (argc > 1) { freopen(argv[1],"r",stdin); }
    ios_base::sync_with_stdio(false);cin.tie(0);
    // PROGRAM STARTS HERE
    ll T; cin >> T;
    for (ll tt=1;tt<=T;tt++) {
        ll N,M; cin >> N >> M; vector<string> l1(N),l2(M); FOR(i,N) cin >> l1[i]; FOR(i,M) cin >> l2[i];
        set<string> dd;
        for (auto &s : l1) {
            dd.insert(s);
            for (ll i=1;i<s.size();i++) { if (s[i] == '/') dd.insert(s.substr(0,i)); }
        }
        ll n1 = dd.size();
        for (auto &s : l2) {
            dd.insert(s);
            for (ll i=1;i<s.size();i++) { if (s[i] == '/') dd.insert(s.substr(0,i)); }
        }
        ll n2 = dd.size();
        auto ans = n2-n1;
        printf("Case #%lld: %lld\n",tt,ans);
    }
}

